src/Application.js

```diff
@@ -173,6 +155,53 @@ export class Application {
     });
   }
 
+  /**
+   * preloads, initializes and starts plugins.
+   * @return {void}
+   * @memberof Application
+   */
+  setupPlugins() {
+    const preloads = [];
+    Application._plugins.forEach(plugin => {
+      if (!plugin.preload) {
+        return;
+      }
+
+      preloads.push(
+        plugin.preload(this).catch(function preloadFail(error) {
+          plugin.preloadFailed = true;
```

Comment:
For consistency, maybe we should go ahead and set `preloadFailed` to false on the plugin before we start the preload:

```
plugin.preloadFailed = false;

preloads.push(
...
```

Reply:
That's actually only used in the loading process, failed plugins get removed from the list, with a warning message.